chainer/links/model/vision/googlenet.py

```diff
@@ -0,0 +1,401 @@
+    available = False
+    _import_error = e
+
+from chainer.dataset.convert import concat_examples
+from chainer.dataset import download
+from chainer import flag
+from chainer.functions.activation.relu import relu
+from chainer.functions.activation.softmax import softmax
+from chainer.functions.array.reshape import reshape
+from chainer.functions.math.sum import sum as sum_function
+from chainer.functions.noise.dropout import dropout
+from chainer.functions.normalization.local_response_normalization import (
+    local_response_normalization)
+from chainer.functions.pooling.average_pooling_2d import average_pooling_2d
+from chainer.functions.pooling.max_pooling_2d import max_pooling_2d
+from chainer.initializers import constant
+from chainer.initializers import uniform
+from chainer import link
+from chainer.links.connection.convolution_2d import Convolution2D
+from chainer.links.connection.inception import Inception
+from chainer.links.connection.linear import Linear
+from chainer.serializers import npz
+from chainer.utils import imgproc
+from chainer.variable import Variable
+
+
+class GoogLeNet(link.Chain):
+
+    """A pre-trained GoogLeNet model provided by BVLC [1].
+
+    When you specify the path of the pre-trained chainer model serialized as
+    a ``.npz`` file in the constructor, this chain model automatically
+    initializes all the parameters with it.
+    This model would be useful when you want to extract a semantic feature
+    vector per image, or fine-tune the model on a different dataset.
+
+    If you want to manually convert the pre-trained caffemodel to a chainer
+    model that can be specified in the constructor,
+    please use ``convert_caffemodel_to_npz`` classmethod instead.
+
+    .. [1] `<https://github.com/BVLC/caffe/tree/master/models/bvlc_googlenet>`_
+
+    Args:
+        pretrained_model (str): the destination of the pre-trained
+            chainer model serialized as a ``.npz`` file.
+            If this argument is specified as ``auto``,
+            it automatically downloads the caffemodel from the internet.
+            Note that in this case the converted chainer model is stored
+            on ``$CHAINER_DATASET_ROOT/pfnet/chainer/models`` directory,
+            where ``$CHAINER_DATASET_ROOT`` is set as
+            ``$HOME/.chainer/dataset`` unless you specify another value
+            as a environment variable. The converted chainer model is
+            automatically used from the second time.
+            If the argument is specified as ``None``, all the parameters
+            are not initialized by the pre-trained model, but the default
+            initializer used in the original paper, i.e.,
+            ``chainer.initializers.GlorotUniform(scale=1.0)``.
+
+    Attributes:
+        available_layers (list of str): The list of available layer names
+            used by ``__call__`` and ``extract`` methods.
+
+    """
+
+    image_mean = numpy.array([104, 117, 123], dtype=numpy.float32)  # BGR
```

Comment:
> In this case, to reduce the size and IO access, I think it would be better to store the image with uint8 into hdf5

Thank you for sharing your example code. I've been also doing the similar way (although I'm using `fuel` instead of chainer's dataset & iterator) and therefore I cannot subtract beferehand the mean pixel values, which are float value in general. In my observation, when the storage is HDD, IO bottleneck is problematic. However, when I use SSD, CPU bottleneck is dominant (especially jpg decoding and resizing).

Anyway, I do not use `image_mean` this time. As you pointed out, it should be added in another PR.